### ISSUE TITLE
CA1508 reported incorrectly

### ISSUE THREAD

**Issue by GiottoVerducci** [dotnet/roslyn-analyzers#6914]
### Analyzer

**Diagnostic ID**: CA1508 'retries == 0' is always 'false'. Remove or refactor the condition(s) to avoid dead code.

### Analyzer source

VS Version 17.7.3
.NET 7.0

### Describe the bug

The following code emits a CA1508 warning:

 ```
       public static async Task DoRetryAsync(this object element)
 {
     var retries = 3;
     while (--retries >= 0)
     {
         try
         {
             element.ToString(); // dummy action
             return;
         }
         catch (Exception)
         {
             if (retries == 0)
             {
                 throw;
             }

             await Task.Delay(200);
         }
     }
 }

```

### Expected behavior
No warning should be emitted.

### Actual behavior
A warning is emitted.

## Additional context
During the last loop, `--retries` evaluates to 0 which is `>= 0`, and the body is entered. If an exception is caught, retries is equal to 0 inside the catch block.

**Comment by mavasani** [comment]
No repro on .NET 8 RC2 SDK
